.github/workflows/staging-deployment.yml

```diff
@@ -124,133 +109,44 @@ jobs:
-          echo "[INFO] Verify new WAR file..."
-          $SSH '
-            if [ ! -f /opt/artemis/artemis.war.new ]; then
-              echo "Error: No WAR file found at /opt/artemis/artemis.war.new"
-              exit 1
-            fi
-          '
-          
-          # Stop Artemis-Service on node
-          echo "[INFO] Stop artemis.service ..."
-          $SSH sudo systemctl stop artemis
-          
-          # Replace old artemis.war
-          echo "[INFO] Rename old artemis.war ..."
-          $SSH mv /opt/artemis/artemis.war /opt/artemis/artemis.war.old || true
-          echo "[INFO] Rename new artemis.war ..."
-          $SSH mv /opt/artemis/artemis.war.new /opt/artemis/artemis.war
-          
-          # Start Artemis-Service on node
-          echo "[INFO] Start artemis.service ..."
-          $SSH sudo systemctl start artemis
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
 
-      - name: Verify Primary Node Deployment
-        id: verify_deployment
-        timeout-minutes: 10
+      - name: Install Artemis-Ansible dependencies
         run: |
-          while true; do
-            echo "Performing health check..."
-          
-            RESPONSE=$(curl -s -f $HEALTH_CHECK_URL || echo '{"status":"DOWN"}')
-            STATUS=$(echo $RESPONSE | grep -o '"status":"[^"]*"' | cut -d'"' -f4)
-          
-            if [ "$STATUS" = "UP" ]; then
-              echo "Health check passed! Application is UP"
-              exit 0
-            else
-              echo "Health check failed. Status: $STATUS"
-              echo "Waiting 10 seconds before next attempt..."
-              sleep 10
-            fi
-          done
+          pip3 install -r requirements.txt
+          ansible-galaxy collection install -r requirements.yml --force
+          ansible-galaxy install -r requirements.yml --force
+          ansible-galaxy install -r ~/.ansible/collections/ansible_collections/ls1intum/artemis/requirements.yml
 
-      - name: Phase 2 - Deploy to Secondary Nodes
+      - name: Set Vault token
         run: |
-          HOSTS_SPACE_SEPARATED=$(echo "$DEPLOYMENT_HOSTS_SECONDARY" | tr -d '\r' | tr '\n' ' ' | awk '{$1=$1};1')
-          WAR_FILE=$(ls -1 artifacts/*.war | head -n 1)
-          
-          echo "Debug: Hosts list: $HOSTS_SPACE_SEPARATED"
-
-          for node in $HOSTS_SPACE_SEPARATED
-          do
-            echo "##################################################################################################"
-            echo "[INFO] Deploy on $node ..."
-            echo "##################################################################################################"
-
-            # Build SSH-command
-            SSH="ssh -o LogLevel=ERROR -i ~/.ssh/id_rsa -l $DEPLOYMENT_USER $node"
+          echo "${{ secrets.VAULT_TOKEN }}" > ~/.vault-token
+          export VAULT_TOKEN=$(cat ~/.vault-token)
```

Comment:
Please make sure to delete this file after the workflow run

Reply:
Done